src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/ioc/PaymentsService.java

```diff
@@ -143,19 +148,31 @@ private CardPaymentRequest buildPaymentRequest(
             .build();
     }
 
-    private Payment from(PaymentDto paymentDto, String nextUrlCurrent) {
+    private Payment from(PaymentDto paymentDto, String nextUrlCurrent, String originalDateCreated) {
         String dateCreated = Optional.ofNullable(paymentDto.getDateCreated())
             .map(date -> date.toLocalDate().toString())
-            .orElse(null);
+            .orElse(originalDateCreated);
+
         String nextUrl = Optional.ofNullable(paymentDto.getLinks().getNextUrl())
             .map(url -> url.getHref().toString())
             .orElse(nextUrlCurrent);
+
+        String feeId = null;
+        if (paymentDto.getFees() != null) {
+            feeId = Arrays.stream(paymentDto.getFees())
+                .map(FeeDto::getId)
```

Comment:
@daveoh was this confirmed that for Payment Fees ID we will use the "id" and not the "code" field? As remember during refinement there was some discussions around that "id" might not be useful as it is just the database ID?

Reply:
Steve Liddiard confirmed in https://tools.hmcts.net/jira/browse/ROC-7723?focusedCommentId=750617&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-750617